### ISSUE TITLE
[Feat] Support dark theme

### ISSUE THREAD

**Issue by marisn** [OSGeo/grass#591]
As there is this new fad of turning everything dark, we could ride the wave while it's all hype. This could be the eye candy required to help sell GRASS 8.0

On a user request I was testing GRASS with dark theme — only minor adjustments are needed. Still, proper dark theme support is possible only with >=wxPython-4.1  + =>wxWidgets-3.1.3, as it starts to provide wx.SystemAppearances with IsDark test to see if dark theme should be enabled automatically.

**Comment by petrasovaa** [comment]
You are talking about system-level dark theme, right? And just detecting that and tweaking some components to look good. wxPython 4.1 will be widespread by then, so that shouldn't be a big problem. Another thing would be to implement our own dark theme, that would be more work I think.

**Comment by marisn** [comment]
> You are talking about system-level dark theme, right? And just detecting that and tweaking some components to look good. wxPython 4.1 will be widespread by then, so that shouldn't be a big problem. Another thing would be to implement our own dark theme, that would be more work I think.

Yes, I would go for system dark theme support. At the moment there are a few issues as in most of places UI colours are not hardcoded and thus dark theme works almost out of box.

**Comment by hellik** [comment]
would be a nice one

**Comment by wenzeslaus** [comment]
This would be great. It is more than an eye candy, more like eye protection. Now is the time to do it (it was not [9 years ago](http://trac.wxwidgets.org/ticket/13377) :-).

**Comment by petrasovaa** [comment]
> Yes, I would go for system dark theme support. At the moment there are a few issues as in most of places UI colours are not hardcoded and thus dark theme works almost out of box.

@marisn, maybe you could update the description of this feature with more detailed analysis/observations of the problems? Possibly bugs to report to wxPython?

**Comment by petrasovaa** [comment]
See progress in #751.

**Comment by wenzeslaus** [comment]
I took the liberty to add a TODO list with what I was able to identify. Feel free to open separate issues (and PRs of course).
